eventmesh-common/src/main/java/org/apache/eventmesh/common/file/WatchFileTask.java

```diff
@@ -55,11 +56,21 @@ public WatchFileTask(String directoryPath) {
             throw new IllegalArgumentException("must be a file directory : " + directoryPath);
         }
 
-        try (WatchService watchService = FILE_SYSTEM.newWatchService()) {
-            this.watchService = watchService;
+        try {
+            this.watchService = FILE_SYSTEM.newWatchService();
+        } catch (IOException ex) {
+            throw new RuntimeException("WatchService initialization fail", ex);
+        }
+
+        try {
             path.register(this.watchService, StandardWatchEventKinds.OVERFLOW, StandardWatchEventKinds.ENTRY_MODIFY,
                 StandardWatchEventKinds.ENTRY_CREATE, StandardWatchEventKinds.ENTRY_DELETE);
-        } catch (Exception ex) {
+        } catch (IOException ex) {
+            try {
+                this.watchService.close();
```

Comment:
The `watchService` created in `WatchFileTask` will not close until the EventMesh server shutdown now. The `WatchFileTask` instances are few and I think it's fine.

Reply:
Are you saying this does not to be closed here?

Reply:
@Pil0tXia @pandaapo Is [this commit](https://github.com/apache/eventmesh/pull/4716/commits/989c9629065e0f2f1678d48ca7ab29d326632bf1) what you had in mind?

Reply:
Please see my latest https://github.com/apache/eventmesh/pull/4716#pullrequestreview-1811550992